### PR TITLE
Fix Reliability/Security reported by SonnarCloud - master

### PR DESCRIPTION
**Pull request by PeterSuna**
- Fix bugs
  - false-positive:
    - https://sonarcloud.io/project/issues?id=PantheonTechnologies_lighty-core&open=AXjzmqKdvlgF324uXqgI&resolved=false&types=BUG
    - https://sonarcloud.io/project/issues?id=PantheonTechnologies_lighty-core&open=AXpXM7qPvpRNvHWtI-3N&resolved=false&types=BUG
    - https://sonarcloud.io/project/issues?id=PantheonTechnologies_lighty-core&open=AXpXM7qPvpRNvHWtI-3O&resolved=false&types=BUG
    - https://sonarcloud.io/project/issues?id=PantheonTechnologies_lighty-core&open=AXrtfaA0uyKPJZF507ly&resolved=false&types=BUG
    - https://sonarcloud.io/project/issues?id=PantheonTechnologies_lighty-core&open=AXjzmqBKvlgF324uXqcR&resolved=false&types=BUG
    - https://sonarcloud.io/project/issues?id=PantheonTechnologies_lighty-core&open=AXpXM7hevpRNvHWtI-z4&resolved=false&types=BUG
    - https://sonarcloud.io/project/issues?id=PantheonTechnologies_lighty-core&open=AXpXM7h0vpRNvHWtI-0I&resolved=false&types=BUG 
- Fix Vulnerability
  - false-positive:
    - https://sonarcloud.io/project/issues?id=PantheonTechnologies_lighty-core&open=AXpXM7xsvpRNvHWtI-4i&resolved=false&types=VULNERABILITY (Encrypted key salt is generated as is recommended) 
    - https://sonarcloud.io/project/issues?id=PantheonTechnologies_lighty-core&open=AXeQ_hp7U76N_IOMmM0m&resolved=false&types=VULNERABILITY (Encrypted key salt is generated as is recommended) 
    - https://sonarcloud.io/project/issues?id=PantheonTechnologies_lighty-core&open=AWvcb0zdKPT9GnU7QGkh&resolved=false&types=VULNERABILITY (Logging username which requested for authorization)
- Fix Security Hotspot
  - false-positive:
    - https://sonarcloud.io/project/security_hotspots?id=PantheonTechnologies_lighty-core&branch=master&resolved=false&types=SECURITY_HOTSPOT  

Signed-off-by: Peter Suna <peter.suna@pantheon.tech>

### CODE REVIEW

File: lighty-modules/lighty-gnmi/lighty-gnmi-device-simulator/src/main/java/io/lighty/modules/gnmi/simulatordevice/gnmi/GnmiCapabilitiesService.java [hunk 1]
```diff
@@ -43,10 +46,12 @@ public GnmiCapabilitiesService(final EffectiveModelContext schemaContext,
             final Gnmi.ModelData.Builder builder = Gnmi.ModelData.newBuilder()
                 .setOrganization(module.getOrganization().orElse("UNKNOWN-ORGANIZATION"))
                 .setName(module.getName());
-            if (module.getSemanticVersion().isPresent()) {
-                builder.setVersion(module.getSemanticVersion().get().toString());
-            } else if (module.getRevision().isPresent()) {
-                builder.setVersion(module.getRevision().get().toString());
+            Optional<SemVer> semanticVersion = module.getSemanticVersion();
+            Optional<Revision> revision = module.getRevision();
```
Comment (vainyksi) [inline lighty-modules/lighty-gnmi/lighty-gnmi-device-simulator/src/main/java/io/lighty/modules/gnmi/simulatordevice/gnmi/GnmiCapabilitiesService.java:50]: I think we do not need this variable

File: lighty-examples/lighty-controller-springboot-netconf/src/main/java/io/lighty/core/controller/springboot/rest/TopologyRestService.java [hunk 2]
```diff
@@ -64,14 +65,16 @@ public ResponseEntity getAllTopologyIdsOperational(Authentication authentication
             final Optional<NetworkTopology> readData =
                     tx.read(LogicalDatastoreType.OPERATIONAL, iid).get(TIMEOUT, TimeUnit.SECONDS);
 
-            if (readData.isPresent() && readData.get().getTopology() != null) {
-                final List<String> topology = readData.get().getTopology().values().stream()
-                        .map(topology1 -> topology1.getTopologyId().getValue())
-                        .collect(Collectors.toList());
-                return ResponseEntity.ok(topology);
-            } else {
-                return ResponseEntity.ok(Collections.emptyList());
```
Comment (vainyksi) [inline lighty-examples/lighty-controller-springboot-netconf/src/main/java/io/lighty/core/controller/springboot/rest/TopologyRestService.java:73]: What is the problem here? I do not see any

File: lighty-core/lighty-codecs/src/main/java/io/lighty/codecs/SerializeIdentifierCodec.java [hunk 3]
```diff
@@ -63,8 +63,10 @@ public YangInstanceIdentifier serialize(final String identifier) {
         DataSchemaContextNode<?> schemaNode = this.dataSchemaContextTree.getRoot();
         for (final String args : pathArgs) {
             final QName qName = getQname(qNameModule, args);
-            if (schemaNode != null && schemaNode.getChild(qName) != null && schemaNode.getChild(qName).isMixin()) {
+            if (schemaNode != null) {
                 schemaNode = schemaNode.getChild(qName);
+            }
+            if (schemaNode != null && schemaNode.isMixin()) {
```
Comment (vainyksi) [inline lighty-core/lighty-codecs/src/main/java/io/lighty/codecs/SerializeIdentifierCodec.java:69]: this is less readable
the class is deprecated - will be removed in next release

File: lighty-modules/lighty-gnmi/lighty-gnmi-sb/src/main/java/io/lighty/gnmi/southbound/device/connection/ConfigurableParameters.java [hunk 4]
```diff
@@ -62,14 +65,17 @@ public ConfigurableParameters(final ExtensionsParameters extensionsParameters) {
...
-            return Optional.of(forceCapabilities.getForceCapability()
-                .entrySet()
-                .stream()
-                .map(model -> model.getValue())
-                .map(model -> Gnmi.ModelData.newBuilder()
-                    .setName(model.getName())
-                    .setVersion(model.getVersion().getValue()).build())
-                .collect(Collectors.toList()));
+            final Map<ForceCapabilityKey, ForceCapability> forceCapability = forceCapabilities.getForceCapability();
+            if (forceCapability != null) {
```
Comment (vainyksi) [inline lighty-modules/lighty-gnmi/lighty-gnmi-sb/src/main/java/io/lighty/gnmi/southbound/device/connection/ConfigurableParameters.java:69]: merge if statements

File: lighty-modules/lighty-gnmi/lighty-gnmi-sb/src/main/java/io/lighty/gnmi/southbound/schema/loader/impl/ByPathYangLoaderService.java [hunk 5]
```diff
@@ -57,8 +57,11 @@
 
     public ByPathYangLoaderService(final Path yangsPath, @Nullable final CrossSourceStatementReactor customReactor) {
         this.yangsPath = Objects.requireNonNull(yangsPath);
-        this.yangParser = new YangParserFactoryImpl(Objects.requireNonNullElse(customReactor,
-                SchemaConstants.DEFAULT_YANG_REACTOR)).createParser();
```
Comment (vainyksi) [inline lighty-modules/lighty-gnmi/lighty-gnmi-sb/src/main/java/io/lighty/gnmi/southbound/schema/loader/impl/ByPathYangLoaderService.java:61]: I do not see any problem here

**Review by vainyksi — CHANGES_REQUESTED** [review]

File: lighty-modules/lighty-gnmi/lighty-gnmi-sb/src/main/java/io/lighty/gnmi/southbound/schema/loader/impl/ByPathYangLoaderService.java (hunk 5 above)
Comment (PeterSuna) [reply lighty-modules/lighty-gnmi/lighty-gnmi-sb/src/main/java/io/lighty/gnmi/southbound/schema/loader/impl/ByPathYangLoaderService.java]: done, marked as false-positive

File: lighty-modules/lighty-gnmi/lighty-gnmi-sb/src/main/java/io/lighty/gnmi/southbound/device/connection/ConfigurableParameters.java (hunk 4 above)
Comment (PeterSuna) [reply lighty-modules/lighty-gnmi/lighty-gnmi-sb/src/main/java/io/lighty/gnmi/southbound/device/connection/ConfigurableParameters.java]: done

File: lighty-core/lighty-codecs/src/main/java/io/lighty/codecs/SerializeIdentifierCodec.java (hunk 3 above)
Comment (PeterSuna) [reply lighty-core/lighty-codecs/src/main/java/io/lighty/codecs/SerializeIdentifierCodec.java]: ok, reverted

File: lighty-examples/lighty-controller-springboot-netconf/src/main/java/io/lighty/core/controller/springboot/rest/TopologyRestService.java (hunk 2 above)
Comment (PeterSuna) [reply lighty-examples/lighty-controller-springboot-netconf/src/main/java/io/lighty/core/controller/springboot/rest/TopologyRestService.java]: done, marked as false-positive

File: lighty-modules/lighty-gnmi/lighty-gnmi-device-simulator/src/main/java/io/lighty/modules/gnmi/simulatordevice/gnmi/GnmiCapabilitiesService.java (hunk 1 above)
Comment (PeterSuna) [reply lighty-modules/lighty-gnmi/lighty-gnmi-device-simulator/src/main/java/io/lighty/modules/gnmi/simulatordevice/gnmi/GnmiCapabilitiesService.java]: done, marked as false-positive

File: lighty-examples/lighty-controller-springboot-netconf/src/main/java/io/lighty/core/controller/springboot/services/UserAccessServiceImpl.java [hunk 6]
```diff
@@ -30,11 +30,11 @@ public UserAccessServiceImpl() {
     public Optional<UserData> login(String sessionId, LoginRequest loginRequest) {
         UserData userData = users.get(loginRequest.getUserName());
         if (userData != null && userData.verifyPassword(loginRequest.getPassword())) {
-            LOG.info("login OK: {} {}", sessionId, loginRequest.getUserName());
+            LOG.info("login OK: {}", sessionId);
```
Comment (vainyksi) [inline lighty-examples/lighty-controller-springboot-netconf/src/main/java/io/lighty/core/controller/springboot/services/UserAccessServiceImpl.java:33]: the username is also helpful in logs for this example

File: lighty-examples/lighty-controller-springboot-netconf/src/main/java/io/lighty/core/controller/springboot/services/UserAccessServiceImpl.java [hunk 7]
```diff
@@ -30,11 +30,11 @@ public UserAccessServiceImpl() {
...
             sessions.put(sessionId, userData);
             return Optional.of(userData);
         }
-        LOG.info("login Failed: {} {}", sessionId, loginRequest.getUserName());
+        LOG.info("login Failed: {}", sessionId);
```
Comment (vainyksi) [inline lighty-examples/lighty-controller-springboot-netconf/src/main/java/io/lighty/core/controller/springboot/services/UserAccessServiceImpl.java:37]: the username is also helpful in logs for this example

**Review by vainyksi — CHANGES_REQUESTED** [review]

File: lighty-examples/lighty-controller-springboot-netconf/src/main/java/io/lighty/core/controller/springboot/services/UserAccessServiceImpl.java (hunk 6 above)
Comment (PeterSuna) [reply lighty-examples/lighty-controller-springboot-netconf/src/main/java/io/lighty/core/controller/springboot/services/UserAccessServiceImpl.java]: done

File: lighty-examples/lighty-controller-springboot-netconf/src/main/java/io/lighty/core/controller/springboot/services/UserAccessServiceImpl.java (hunk 7 above)
Comment (PeterSuna) [reply lighty-examples/lighty-controller-springboot-netconf/src/main/java/io/lighty/core/controller/springboot/services/UserAccessServiceImpl.java]: done

**Review by vainyksi — APPROVED** [review]

**Pull request merged**